middleware/ordered_group.go

```diff
@@ -260,18 +282,26 @@ func (s *relativeOrder) Clear() {
 	s.order = s.order[0:0]
 }
 
-func (s *relativeOrder) insert(i int, id string, pos RelativePosition) error {
+func (s *relativeOrder) insert(i int, pos RelativePosition, ids ...string) error {
 	switch pos {
 	case Before:
-		s.order = append(s.order, "")
-		copy(s.order[i+1:], s.order[i:])
-		s.order[i] = id
-
+		n := len(ids)
+		var src []string
+		if n <= cap(s.order)-len(s.order) {
+			s.order = s.order[:len(s.order)+n]
+			src = s.order
+		} else {
+			src = s.order
+			s.order = make([]string, len(s.order)+n)
+			copy(s.order[:i], src[:i]) // only when allocating a new slice do we need to copy the front half
+		}
+		copy(s.order[i+n:], src[i:])
+		copy(s.order[i:], ids)
 	case After:
 		if i == len(s.order)-1 {
-			s.order = append(s.order, id)
+			s.order = append(s.order, ids...)
 		} else {
-			s.order = append(s.order[:i+1], append([]string{id}, s.order[i+1:]...)...)
+			s.order = append(s.order[:i+1], append(ids, s.order[i+1:]...)...)
```

Comment:
Would be good to have a test that asserts this does the correct behavior if `ids...` is an empty slice.